File: app/src/main/java/com/nidhikamath/kotlinmusicplayerdemo/musicjava/Music.java
package com.nidhikamath.kotlinmusicplayerdemo.musicjava;

import android.content.ContentUris;
import android.net.Uri;

public class Music {
    long id;
    String artist;
    String title;
    String album;
    long albumId;
    long duration;

    public Music(long id, String artist, String title, String album, long duration, long albumId) {
        this.id = id;
        this.artist = artist;
        this.title = title;
        this.album = album;
        this.duration = duration;
        this.albumId = albumId;
    }

    public long getId() {
        return id;
    }

    public String getArtist() {
        return artist;
    }

    public String getTitle() {
        return title;
    }

    public String getAlbum() {
        return album;
    }

    public long getDuration() {
        return duration;
    }

    public long getAlbumId() {
        return albumId;
    }

    public Uri getURI() {
        return ContentUris.withAppendedId(
                android.provider.MediaStore.Audio.Media.EXTERNAL_CONTENT_URI, id);
    }

    public Uri getAlbumArt() {
        try {
            Uri genericArtUri = Uri.parse("content://media/external/audio/albumart");
            Uri actualArtUri = ContentUris.withAppendedId(genericArtUri, albumId);
            return actualArtUri;
        } catch(Exception e) {
            return null;
        }
    }
}

File: app/src/main/java/com/nidhikamath/kotlinmusicplayerdemo/helper/Music.kt
package com.nidhikamath.kotlinmusicplayerdemo.helper

import android.content.ContentUris
import android.net.Uri
import android.provider.MediaStore

class Music(
    var id: Long,
    var artist: String,
    var title: String,
    var album: String,
    var duration: Long,
    var albumId: Long
){
    fun getURI(): Uri {
        return ContentUris.withAppendedId(
            MediaStore.Audio.Media.EXTERNAL_CONTENT_URI, id
        )
    }

    fun getAlbumArt(): Uri? {
        return try {
            val genericArtUri = Uri.parse("content://media/external/audio/albumart")
            ContentUris.withAppendedId(genericArtUri, albumId)
        } catch (e: Exception) {
            null
        }
    }
}


